components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/internal/TemplateManagerDataHolder.java

```diff
@@ -41,12 +68,56 @@ public void setConfigurationManager(ConfigurationManager configurationManager) {
     }
 
     /**
-     * Get the ConfigurationManager object held at the data holder.
+     * Get read write template handler.
      *
-     * @return configuration manger object.
+     * @return template handler.
      */
-    public ConfigurationManager getConfigurationManager() {
+    public TemplateHandler getReadWriteTemplateHandler() {
 
-        return this.configurationManager;
+        return readWriteTemplateHandler;
+    }
+
+    /**
+     * Set read write template handler.
+     *
+     * @param readWriteTemplateHandler template handler.
+     */
+    public void setReadWriteTemplateHandler(TemplateHandler readWriteTemplateHandler) {
+
+        this.readWriteTemplateHandler = readWriteTemplateHandler;
+    }
+
+    /**
+     * Get template handlers.
+     *
+     * @return list of template handlers.
+     */
+    public List<ReadOnlyTemplateHandler> getReadOnlyTemplateHandlers() {
+
+        return Collections.unmodifiableList(readOnlyTemplateHandlers);
+    }
+
+    /**
+     * Add template handler.
+     *
+     * @param templateHandler template handler.
+     */
+    public void addReadOnlyTemplateHandler(ReadOnlyTemplateHandler templateHandler) {
+
+        this.readOnlyTemplateHandlers.add(templateHandler);
+    }
+
+    /**
+     * Remove template handler.
+     *
+     * @param templateHandler template handler.
+     */
+    public void removeReadOnlyTemplateHandler(ReadOnlyTemplateHandler templateHandler) {
+
+        this.readOnlyTemplateHandlers.remove(templateHandler);
+    }
+
+    public static Map<String, Template> getFileBasedTemplates() {
```

Comment:
add new line after method declaration

Reply:
addressed in feace15

Reply:
this method should non-static

Reply:
There should be another method to addFileBasedTemplate()

Reply:
This should return return Collections.unmodifiableMap(fileBasedTemplates)